### ISSUE TITLE
Fix doc of `torch.nn.MultiLabelSoftMarginLoss`

### ISSUE THREAD

**Issue by mle-els** [pytorch/pytorch#92000]
### 📚 The doc issue

The documentation states that $y[i] \in$ {0, 1} and later on, it also say "label targets padded by -1 ensuring same shape as the input". This is contradictory. Looking at the source code, I believe the bit about padding is incorrect and -1 values will turn the loss the wrong way.

### Suggest a potential alternative/fix

remove "padded by -1"

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki

**Comment by drisspg** [comment]
If you would like to submit a PR I would be happy to review this

**Comment by mle-els** [comment]
I don't feel I have completely understand this implementation so it's better leaving it to someone more competent.

**Comment by dvorst** [comment]
Indeed, "padded by -1" is incorrect. The shape of input and target tensor are compared, and an exception is raised if they differ in either dimension 0 or 1. See the code snippet below and the resulting output:

```
import torch
import torch.nn as nn

# Create some example data
input = torch.tensor(
    [
        [0.8, 0.2, -0.5],
        [0.1, 0.9, 0.3],
    ]
)
target1 = torch.tensor(
    [
        [1, 0, 1],
        [0, 1, 1],
        [0, 1, 1],
    ]
)
target2 = torch.tensor(
    [
        [1, 0],
        [0, 1],
    ]
)
target3 = torch.tensor(
    [
        [1, 0, 1],
        [0, 1, 1],
    ]
)
loss_func = nn.MultiLabelSoftMarginLoss()
try:
    loss = loss_func(input, target1).item()
except RuntimeError as e:
    print('target1 ', e)
try:
    loss = loss_func(input, target2).item()
except RuntimeError as e:
    print('target2 ', e)
loss = loss_func(input, target3).item()
print('target3 ', loss)
```

output:
```
target1  The size of tensor a (3) must match the size of tensor b (2) at non-singleton dimension 0
target2  The size of tensor a (2) must match the size of tensor b (3) at non-singleton dimension 1
target3  0.6305370926856995
```

Creating PR
